addons/website_event/static/src/js/website_event.js

```diff
@@ -41,26 +45,29 @@ var EventRegistrationForm = publicWidget.Widget.extend({
      * @param {Event} ev
      */
     on_click: function (ev) {
+        var self = this;
         ev.preventDefault();
         ev.stopPropagation();
-        var $form = $(ev.currentTarget).closest('form');
-        var $button = $(ev.currentTarget).closest('[type="submit"]');
+        const formEl = ev.currentTarget.closest("form");
+        const buttonEl = ev.currentTarget.closest("[type='submit']");
         const post = this._getPost();
-        $button.attr('disabled', true);
-        return rpc($form.attr('action'), post).then(function (modal) {
-            var $modal = $(modal);
-            $modal.find('.modal-body > div').removeClass('container'); // retrocompatibility - REMOVE ME in master / saas-19
-            $modal.appendTo(document.body);
-            const modalBS = new Modal($modal[0], {backdrop: 'static', keyboard: false});
-            modalBS.show();
-            $modal.appendTo('body').modal('show');
-            $modal.on('click', '.js_goto_event', function () {
-                $modal.modal('hide');
-                $button.prop('disabled', false);
-            });
-            $modal.on('click', '.btn-close', function () {
-                $button.prop('disabled', false);
+        buttonEl.disabled = true;
+        return rpc(formEl.action, post).then(function (modal) {
+            if (!self.modalEl) {
+                const modalEl = new DOMParser().parseFromString(modal, "text/html").body.firstChild;
+                modalEl.querySelector(".js_goto_event").addEventListener("click", () => {
+                    buttonEl.disabled = false;
+                });
+                modalEl.querySelector(".btn-close").addEventListener("click", () => {
+                    buttonEl.disabled = false;
+                });
+                self.modalEl = modalEl;
```

Comment:
Caching the modal doesn't work: if you select anything, click on the "Register" button, then cancel and try to select something else (more / less tickets, or a different combination), you'll be stuck with the 1st modal.

_Not_ caching it would also allow to remove the event listeners inside the function triggered in `addEventListener`. e.g.:
```js
const _onClick = () => {
    buttonEl.disabled = false;
    modalEl.querySelector(".js_goto_event").removeEventListener("click", _onClick);
    modalEl.querySelector(".btn-close").removeEventListener("click", _onClick);
};
```

I don't think caching the modal is needed anyway. Was there a specific reason or was it just intended as a quick win? :)

Reply:
> you'll be stuck with the 1st modal

Agree with this point. 

>Was there a specific reason 

Previously on each click it was creating new modal and append it to the DOM. so as result, at a same time there's multiple modal appended in DOM. To avoid that I have cached the modal.

Reply:
I see. Then the cache was not the right solution, but we could remove the modalEl at the end of the `_onClick` event:
```js
const _onClick = () => {
    buttonEl.disabled = false;
    modalEl.querySelector(".js_goto_event").removeEventListener("click", _onClick);
    modalEl.querySelector(".btn-close").removeEventListener("click", _onClick);
    modalEl.remove();
};
```